Assets/Mirror/Components/NetworkSceneChecker.cs

```diff
@@ -0,0 +1,119 @@
+using UnityEngine;
+using System.Collections.Generic;
+using UnityEngine.SceneManagement;
+
+namespace Mirror
+{
+    /// <summary>
+    /// Component that controls visibility of networked objects between scenes.
+    /// <para>Any object with this component on it will only be visible to other objects in the same scene</para>
+    /// <para>This would be used when the server has multiple additive subscenes loaded to isolate players to their respective subscenes</para>
+    /// </summary>
+    [AddComponentMenu("Network/NetworkSceneChecker")]
+    [RequireComponent(typeof(NetworkIdentity))]
+    [HelpURL("https://mirror-networking.com/docs/Components/NetworkSceneChecker.html")]
+    public class NetworkSceneChecker : NetworkBehaviour
+    {
+        /// <summary>
+        /// Flag to force this object to be hidden from all observers.
+        /// <para>If this object is a player object, it will not be hidden for that client.</para>
+        /// </summary>
+        [Tooltip("Enable to force this object to be hidden from all observers.")]
+        public bool forceHidden;
+
+        static Dictionary<string, HashSet<NetworkIdentity>> sceneCheckerObjects;
+
+        [Scene, SerializeField]
+        string currentScene;
```

Comment:
why is this serialized? does this show in inspector?
if yes, why not use scenemanager.activescene?

Reply:
I have it showing in inspector mostly for debugging...doesn't need to be.  scenemanager.activescene is not the same as gameobject.scene when the object is in an additive scene.

Reply:
I agree this should not be serialized.

@vis2k  the use case here is that you have a bunch of scenes loaded at the same time.
scenemanager.activescene  only shows the main scene,  but he is trying to have players segregated in multiple scenes.  So for example:  Suppose he has 5 scene loaded:

1) Main Scene
2) Room 1
     Player 1
     Player 2
3) Room 2
     Player 3
4) Room 3
     Player 4
     Player 5
     Player 6

No player will be in the main scene in this case
player 1 and player 2 should see each other,  but nobody else
player 3 should only see himself
Player 4,5,6 should see each other.

Suppose player 4 moves to Room 1,   his currentScene will change from Room 3 to Room 1,   This field is required because you need to know he left Room 3 and he needs to notify 5 and 6  to rebuild their observers.

Reply:
can't we just use gameobject.scene? this field seems redundant

Reply:
I need to keep track of the scene it was in before I set that to gameobject.scene when gameobject.scene changes.

Reply:
makes sense. just keep it withotu the scene,serializefield then. 
for debugging you can use unity inspector's debug view to still see it

Reply:
Done